pkg/controller/ps/status.go

```diff
@@ -147,8 +176,35 @@ func (r *PerconaServerMySQLReconciler) reconcileCRStatus(ctx context.Context, cr
 		cr.Status.State = apiv1alpha1.StateInitializing
 	}
 
-	if err := r.checkTLSIssuer(ctx, cr); err != nil {
-		cr.Status.State = apiv1alpha1.StateError
+	if cr.Status.State != apiv1alpha1.StateError {
+		if cr.Status.State == apiv1alpha1.StateInitializing {
+			clusterCondition.Type = apiv1alpha1.StateInitializing.String()
+			clusterCondition.Reason = apiv1alpha1.StateInitializing.String()
+			clusterCondition.Status = metav1.ConditionTrue
+			meta.SetStatusCondition(&cr.Status.Conditions, clusterCondition)
+
+			clusterCondition.Type = apiv1alpha1.StateReady.String()
+			clusterCondition.Reason = apiv1alpha1.StateReady.String()
+			clusterCondition.Status = metav1.ConditionFalse
+			meta.SetStatusCondition(&cr.Status.Conditions, clusterCondition)
+		}
+
+		if cr.Status.State == apiv1alpha1.StateReady {
+			clusterCondition.Type = apiv1alpha1.StateReady.String()
+			clusterCondition.Reason = apiv1alpha1.StateReady.String()
+			clusterCondition.Status = metav1.ConditionTrue
+			meta.SetStatusCondition(&cr.Status.Conditions, clusterCondition)
+
+			clusterCondition.Type = apiv1alpha1.StateInitializing.String()
+			clusterCondition.Reason = apiv1alpha1.StateInitializing.String()
+			clusterCondition.Status = metav1.ConditionFalse
+			meta.SetStatusCondition(&cr.Status.Conditions, clusterCondition)
+		}
+
+		// clusterCondition.Type = cr.Status.State.String()
+		// clusterCondition.Reason = cr.Status.State.String()
+
+		// meta.SetStatusCondition(&cr.Status.Conditions, clusterCondition)
 	}
```

Comment:
We can simplify it like this. Maybe the for loop doesn't make it simpler, but at least the switch statement should be there.

```suggestion
	switch cr.Status.State {
	case apiv1alpha1.StateInitializing, apiv1alpha1.StateReady:
		for _, condition := range []apiv1alpha1.StatefulAppState{apiv1alpha1.StateInitializing, apiv1alpha1.StateReady} {
			clusterCondition.Type = condition.String()
			clusterCondition.Reason = condition.String()
			if cr.Status.State == condition {
				clusterCondition.Status = metav1.ConditionTrue
			} else {
				clusterCondition.Status = metav1.ConditionFalse
			}
			meta.SetStatusCondition(&cr.Status.Conditions, clusterCondition)
		}
	}
```

Reply:
Nice @pooknull , like it a lot

Reply:
[a543e7e](https://github.com/percona/percona-server-mysql-operator/pull/543/commits/a543e7eaa8fed42918c9332c476b705bbbbc5e1c)